### PR TITLE
First take at some marketing analytics.

### PR DESCRIPTION
**Pull request by jonathanmarvens**
Added analytics tracking for the promotions.

### CODE REVIEW

**Comment by jonathanmarvens** [comment]
__CC:__ @npm/dubdub, @cowperthwait.

**Comment by cowperthwait** [comment]
@jonathanmarvens I don’t actually understand how this works, so I trust you. At some point when we’re in mutually convenient time zones, want to do a quick 5-min Slack to talk me thru the Google Analytics parts?

File: adapters/bonbon.js [hunk 1]
```diff
@@ -89,6 +90,10 @@ exports.register = function(server, options, next) {
       env: {
         NPMO_COBRAND: process.env.NPMO_COBRAND,
         CANONICAL_HOST: process.env.CANONICAL_HOST
+      },
+      ga: {
+        dimensions: JSON.stringify(qs.parse(process.env.PROMOTION_GA_DIMENSIONS)),
+        id: process.env.GA_ID
```
Comment (aredridel) [inline adapters/bonbon.js]: A fallback to the old ID with `|| 'UA-47041310-1'` here would make this deployable with no new config active.

File: templates/layouts/default.hbs [hunk 2]
```diff
@@ -62,13 +62,15 @@
   <script src="/static/{{m "js/index.min.js"}}"></script>
   {{/if}}
 
+  <meta name="x-npm:ga:id" content="{{{ga.id}}}">
```
Comment (aredridel) [inline templates/layouts/default.hbs]: Yeah, this can be just `{{` not `{{{` -- if there were any escaping to do, we'd want to do it.

**Comment by aredridel** [comment]
This looks pretty good. A couple comments, and Raquel's questions stand!

**Comment by jonathanmarvens** [comment]
@rockbot:

1. The content is coming from our CMS API, but some of the `{{{` usage here was accidental and not what we want, so thanks for catching it!
2. Re `PROMOTION_GA_DIMENSIONS`:
  - Just a query string: `'aboveChrome=dimension1&belowHeader=dimension3&packagePage=dimension2'`.
  - AFAIK, it's fine if anyone sees it.

File: templates/layouts/default.hbs (hunk 2 above)
Comment (jonathanmarvens) [inline templates/layouts/default.hbs]: @aredridel: Fixed. Thanks for catching that!

File: adapters/bonbon.js (hunk 1 above)
Comment (jonathanmarvens) [inline adapters/bonbon.js]: @aredridel: IDK how I feel about that, TBH. If we're gonna go with using an environment variable for this, is the fallback necessary?

Comment (aredridel) [inline adapters/bonbon.js]: Nope! It'll change deployment strategy -- what we have to do to launch this -- but I'm happy with whichever way you decide. What happens if we screw up deployment and get the variable name or value wrong?

**Comment by aredridel** [comment]
Agreed, those values are not private or secret.

File: templates/layouts/default.hbs (hunk 2 above)
Comment (aredridel) [inline templates/layouts/default.hbs]: \o/

File: adapters/bonbon.js (hunk 1 above)
Comment (jonathanmarvens) [inline adapters/bonbon.js]: @aredridel: I understand the reasoning, but I'd prefer to leave it as, unless there's a strong case for going that route. OK?

**Comment by jonathanmarvens** [comment]
@npm/dubdub: OK to merge?

**Comment by aredridel** [comment]
\o/

**Pull request merged**